### DISCUSSION TITLE
making a public link!

### DISCUSSION THREAD

**Discussion by wavuup** [oobabooga/text-generation-webui#1726] [Q&A] [answered]
Im pretty new to computers and was having trouble creating a public link using the oobabooga UI, it gave me the statement to set 'share=true' in 'launch()' however because of my lack of knowledge of the system I wasn't sure how to go about it, ive looked around forms and things of the sort for a couple days and wasn't sure what to do.

**Comment by Felixqian4160** [comment]
use --share in flag

**Reply by wavuup** [reply]
how do i do that? sorry im just very new to using oobabooga

**Reply by mattjaybe** [reply, marked as answer]
Open up webui.py with Notepad++ (or any text editor of choice) and near the bottom find this line:

`run_cmd("python server.py --auto-devices --api --chat --model-menu")`

Add --share to it so it looks like this:

`run_cmd("python server.py --auto-devices --api --chat --model-menu --share")`

You can add any other arguments to it, too. See the main github page for a list of them.

**Reply by ghost** [reply]
you can also pass it as a parameter.. no editing required

**Reply by wavuup** [reply]
Thank you so much :)

**Reply by Arkodomini** [reply]
hi guys i have the same problem and i just downloaded it but i cant find that line in webui.py

**Comment by saintskytower** [comment]
> hi guys i have the same problem and i just downloaded it but i cant find that line in webui.py

Those flags have moved to an external file called CMD_FLAGS.txt found the root of the install directory for WebUI.

**Comment by saifmaxx** [comment]
> hi guys i have the same problem and i just downloaded it but i cant find that line in webui.py

Open CMD_FLAGS.txt  file and just add **_--share_** and then start the program again

**Reply by Mulakulu** [reply]
This doesn't seem to be working on the latest release anymore, and now I can't even find webui.py in the files either. Some better documentation on the developers end would be greatly appreciated, since the program still says "To create a public link, set `share=True` in `launch()`." with absolutely to clear indication on where "launch()" is. They also have "To define persistent command-line flags like `--listen` or `--api`, edit the `CMD_FLAGS.txt`" But that clearly doesn't work

**Reply by ted-phlx** [reply]
Have you gotten this to work yet? I'm stuck having found out pretty much what @Mulakulu pointed out

**Reply by Nekomota** [reply]
Looked in server.py, and found a launch() function (just control-f 'launch') with share as an arg, so i changed it from 
share=shared.args.share
To
share=True
And restarted the webui, and it did create a public link like it used to.

**Comment by ghost** [comment]
I simply pass parameters when i start the server.  works fine for me without having to edit anything. ( well technically my system is passing it for me, as i have it running as a service, but same concept, its all command line parameters.

**Reply by Jinchurukie** [reply]
How did you do this?

**Reply by ted-phlx** [reply]
I've resorted to doing the same thing, with an instance on vast.ai no less. None of the other possibilities worked... so I just set it up that way, as Onstart parameters.

**Reply by ghost** [reply]
Just add --xxx to your command line when you start the server.  In my case its via systemD and i use conda which is activated earlier in my script, but its the same concept. just replace my parameters with the ones you want. I only present to my local network, and use a reverse proxy with login to get outside world access.   So listen + port is all i need

my actual start command ->   "python -u $HOME/src/text-generation-webui/server.py --listen  --listen-port 8080 --threads 32" ( no quotes )

I would think the most simple would be:  "python server.py --share" ( no quotes ) to get the public share going  ( assuming you have activated whatever venv you are using first )
